FLPATH-274 enable spring-cloud-config

sergiubodiu opened the pull request:
I've enabled it for local development where the configserver can be passed through variable or located at default location localhost:8888
For testing purposes created demo config server https://github.com/sergiubodiu/parodos-configserver with configurations copied from local development
https://github.com/sergiubodiu/parodos-config-repo

pkliczewski reviewed (commented):
lgtm

gciavarrini commented on notification-service/src/main/resources/application-local.yml:
```diff
@@ -43,9 +43,10 @@ spring:
     username: postgres
     password: postgres
 
-springdoc:
-  writer-with-order-by-keys: true
-  writer-with-default-pretty-printer: true
```
These are necessary to properly generate SDK files. Why do you delete them?

sergiubodiu replied:
These properties are redundant, they are in application.yml so for local profile it will be inherited from the https://github.com/parodos-dev/parodos/blob/main/notification-service/src/main/resources/application.yml

gciavarrini replied:
You're right! Thanks

pkliczewski commented:
The change looks good to me, please rebase

pkliczewski commented:
/lgtm

sergiubodiu commented:
rebased, please review again.
Fixed flakiness in the test because of multiple json libraries in the build path see https://github.com/spring-projects/spring-boot/issues/9248
Fixed has for new JDK17 dependency CI

pkliczewski commented on workflow-service/src/test/java/com/redhat/parodos/workflow/version/WorkFlowVersionServiceImplTest.java:
```diff
@@ -24,7 +24,7 @@ public void testGetHash() {
 		assertDoesNotThrow(() -> {
 			String hash = this.workFlowVersionService.getHash(object);
-			assertEquals(hash, "b0c8ed039dc102c0bab6a5e979931a0b");
+			assertEquals(hash, "3ffe85fac846ce265bcef0534e6334b7");
```
This change won't be needed after https://github.com/parodos-dev/parodos/pull/239

pkliczewski reviewed (commented):
/lgtm

pkliczewski reviewed (commented):
/lgtm

masayag commented:
@sergiubodiu since the target deployment of parodos is k8s/openshift, what changes will be required to support configmap and secret as configuration resources of the application?
https://github.com/spring-cloud/spring-cloud-kubernetes#spring-cloud-kubernetes-configserver

sergiubodiu commented:
Needs approve label to be merged. @masayag it doesn't require changes on parodies as we're consumer of the configuration. Spring-Cloud-Kubernetes config server is a server side change. Parodos being a client will consume those secrets / config maps and inject it into the variables that we define.

I created a demo server https://github.com/sergiubodiu/parodos-configserver

However if we want to configure dynamic Kubernetes clients we could look into Fabric8 or the next gen https://github.com/eclipse/jkube both have a lot of red hat contributors
https://developers.redhat.com/articles/2023/01/05/how-use-fabric8-kubernetes-client

masayag commented:
> Needs approve label to be merged. @masayag it doesn't require changes on parodos as we're consumer of the configuration. Spring-Cloud-Kubernetes config server is a server side change. Parodos being a client will consume those secrets / config maps and inject it into the variables that we define.
> 
> I created a demo server https://github.com/sergiubodiu/parodos-configserver
> 

Should the config-server be distributed as part of parodos deployment or is the expectation for a user to have its own config-server?

My question is, should we move the parodos-configserver project into parodos organization and use it in our tests when deployed on top of k8s?
In that case, we'll have to build and deploy parodos-configserver image to be applied on k8s. But we'll be able to demo and show how it is working as part of our build process.

> However if we want to configure dynamic Kubernetes clients we could look into Fabric8 or the next gen https://github.com/eclipse/jkube both have a lot of red hat contributors https://developers.redhat.com/articles/2023/01/05/how-use-fabric8-kubernetes-client

masayag commented:
/approve

pkliczewski commented:
@sergiubodiu please rebase and we will merge. All labels are added
